### PR TITLE
OCPQE-17830: Fix set-sample-operator-disconnected on aws-ipi-disconnected-sts-basecap-none profile

### PR DESCRIPTION
**Pull request by wewang58**

### CODE REVIEW

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
now in 4.15 config.samples part it passed, but seems check_mirror_registry function has somethins issues too, checking it  cc @jinyunma

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

**Comment by wewang58** [comment]
@jinyunma  @jianlinliu  could you help to review it, thanks
since image registry is uninstalled, image registry test cases are all skipped according to openshift-tests-private code: 
'''
	g.BeforeEach(func() {
		var message string
		if !checkOptionalOperatorInstalled(oc, "ImageRegistry") {
			g.Skip("Skip for the test due to image registry not installed")
		}
'''
so skip the step about set-sample-operator.

**Comment by jianlinliu** [comment]
> since image registry is uninstalled, image registry test cases are all skipped according to openshift-tests-private code:
> so skip the step about set-sample-operator.

1. Is there any dependency between image-registry and openshift-sample operator? 
2. Actually there is a dedicated optional cap - openshift-samples, which become optional in 4.11, does that have more impact on `set-sample-operator-disconnected` setp

**Comment by jinyunma** [comment]
> why the above 4.14 install can get it passed?

Because similar functions are also running in step `enable-qe-catalogsource-disconnected`, including `set_CA_for_nodes` and `create_settled_icsp`, so step `set-sample-operator-disconnected` can pass on 4.14. 
But on 4.15, OLM becomes operational operator, and step `enable-qe-catalogsource-disconnected` is skipped in this profile as OLM is disabled.

> since image registry is uninstalled, image registry test cases are all skipped according to openshift-tests-private code,  so skip the step about set-sample-operator.

I think it might still have problem when OLM is disabled and image-registry is enabled, maybe also need to consider creating icsp in `enable-qe-catalogsource-disconnected` in such case to avoid error in [failed job](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/45365/rehearse-45365-periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28/1721845411389378560/artifacts/aws-ipi-disconnected-sts-basecap-none-f28/set-sample-operator-disconnected/build-log.txt), right?
```
  ! error: Import failed (InternalError): Internal error occurred: quay.io/openshifttest/busybox@sha256:c5439d7db88ab5423999530349d327b04279ad3161d7596d2126dfb5b02bfd1f: Get "https://quay.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
      Less than a second ago
```

**Comment by wewang58** [comment]
> > since image registry is uninstalled, image registry test cases are all skipped according to openshift-tests-private code:
> > so skip the step about set-sample-operator.
> 
> Is there any dependency between image-registry and openshift-sample operator?
> 
> Compared with a successful 4.14 install [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive/1720764590129680384/artifacts/aws-ipi-disconnected-sts-basecap-none-f28-destructive/set-sample-operator-disconnected/build-log.txt), I think the key difference is from this [line](https://github.com/openshift/release/blob/b74130fa4c2f93af1bdabaf766729a4927a70484/ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh#L76), I wonder the `image.config.openshift.io/cluster` is from image-registry operator install? If that is true, and from OCP 4.14, the ImageRegistry is optional, why the above 4.14 install can get it pa

**Comment by wewang58** [comment]
> > since image registry is uninstalled, image registry test cases are all skipped according to openshift-tests-private code:
> > so skip the step about set-sample-operator.
> 
> Is there any dependency between image-registry and openshift-sample operator?
> 
> Compared with a successful 4.14 install [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive/1720764590129680384/artifacts/aws-ipi-disconnected-sts-basecap-none-f28-destructive/set-sample-operator-disconnected/build-log.txt), I think the key difference is from this [line](https://github.com/openshift/release/blob/b74130fa4c2f93af1bdabaf766729a4927a70484/ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh#L76), I wonder the `image.config.openshift.io/cluster` is from image-registry operator install? If that is true, and from OCP 4.14, the ImageRegistry is optional, why the above 4.14 install can get it passed?

@jianlinliu  commnuicated with @jinyunma , we maybe meet disable olm and enable image-registry scenario, so i will update codes , when ready will let you review again, thanks

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

**Comment by jianlinliu** [comment]
> we maybe meet disable olm and enable image-registry scenario

Actually I think firstly we need to figure out the exact usage of `set-sample-operator-disconnected`. For <4.14, when openshift-sample is disabled, `set-sample-operator-disconnected` still run some extra actions, including `set_CA_for_nodes` and `mirror_tag_images`, what is the purpose for that? For my understanding, if openshift-sample is disabled, `set-sample-operator-disconnected` should be skipped.

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
@jialiu @jinyunma  i think if image registry disable, we can also skip the set-sample-operator step, now i simulate olm disable and image-registy enable,[ the 4.15 job passed](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/45365/rehearse-45365-periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28/1722501829972987904), how about we add comments and update the step name to make it more clear, thanks @jianzhangbjz  could you help to review too, thanks

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

**Comment by wewang58** [comment]
the failed job is not related the step, just failed in test cases part

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-disconnected-sts-ep-fips-f28-destructive

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-usgov-ipi-disconnected-private-f14

File: ci-operator/step-registry/mirror-images/disconnected/mirror-images-disconnected-commands.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+set_proxy
+run_command "oc whoami"
+run_command "oc version -o yaml"
+
+# vmc.mirror-registry.qe.devcluster.openshift.com:5000
+MIRROR_PROXY_REGISTRY=`head -n 1 "${SHARED_DIR}/mirror_registry_url"`
+echo "MIRROR_PROXY_REGISTRY: ${MIRROR_PROXY_REGISTRY}"
+# When mirror images using `oc image mirror` command, need unset proxies
+unset_proxy
```
Comment (jianlinliu) [inline ci-operator/step-registry/mirror-images/disconnected/mirror-images-disconnected-commands.sh:66]: I think we do not need the function here, also the following set_proxy

File: ci-operator/step-registry/mirror-images/disconnected/mirror-images-disconnected-commands.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    # optional_auth=`echo -n "${optional_auth_user}:${optional_auth_password}" | base64 -w 0`
+    
+    registry_cred=`head -n 1 "/var/run/vault/mirror-registry/registry_creds" | base64 -w 0`
+    jq --argjson a "{\"${MIRROR_PROXY_REGISTRY}\": {\"auth\": \"$registry_cred\"}}" '.auths |= . + $a' "${CLUSTER_PROFILE_DIR}/pull-secret" > /tmp/new-dockerconfigjson
+    run_command "oc image mirror -f \"/tmp/tag_images_list\"  --insecure=true -a \"/tmp/new-dockerconfigjson\" --skip-missing=true --skip-verification=true --keep-manifest-list=true --filter-by-os='.*'"
+}
+
+set_proxy
+run_command "oc whoami"
+run_command "oc version -o yaml"
```
Comment (jianlinliu) [inline ci-operator/step-registry/mirror-images/disconnected/mirror-images-disconnected-commands.sh:60]:
```suggestion
run_command "oc version --client"
```

File: ci-operator/step-registry/mirror-images/disconnected/mirror-images-disconnected-commands.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    # optional_auth_password=$(cat "/var/run/vault/mirror-registry/registry_quay.json" | jq -r '.password')
+    # optional_auth=`echo -n "${optional_auth_user}:${optional_auth_password}" | base64 -w 0`
+    
+    registry_cred=`head -n 1 "/var/run/vault/mirror-registry/registry_creds" | base64 -w 0`
+    jq --argjson a "{\"${MIRROR_PROXY_REGISTRY}\": {\"auth\": \"$registry_cred\"}}" '.auths |= . + $a' "${CLUSTER_PROFILE_DIR}/pull-secret" > /tmp/new-dockerconfigjson
+    run_command "oc image mirror -f \"/tmp/tag_images_list\"  --insecure=true -a \"/tmp/new-dockerconfigjson\" --skip-missing=true --skip-verification=true --keep-manifest-list=true --filter-by-os='.*'"
+}
+
+set_proxy
+run_command "oc whoami"
```
Comment (jianlinliu) [inline ci-operator/step-registry/mirror-images/disconnected/mirror-images-disconnected-commands.sh:59]: I think we do not need the above two lines

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh [hunk 4]
```diff
@@ -141,24 +91,18 @@ function check_mirror_registry () {
...
 MIRROR_PROXY_REGISTRY=`head -n 1 "${SHARED_DIR}/mirror_registry_url"`
 echo "MIRROR_PROXY_REGISTRY: ${MIRROR_PROXY_REGISTRY}"
 set_CA_for_nodes
-# config_mirror_auth
-# When mirror images using `oc image mirror` command, need unset proxies
-unset_proxy
-mirror_tag_images
-set_proxy
-if [[ $ret -eq 0 ]]; then
-    config_samples_operator
```
Comment (jianlinliu) [inline ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:162]: We still need the `config_samples_operator` here, right?

File: ci-operator/step-registry/mirror-images/disconnected/mirror-images-disconnected-ref.metadata.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+	"path": "mirror-images/disconnected/mirror-images-disconnected-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/mirror-images/disconnected/mirror-images-disconnected-ref.metadata.json:2]:
```suggestion
	"path": "mirror-images/tag_images/mirror-images-tag_images-ref.yaml",
```

File: ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh [hunk 6]
```diff
@@ -430,6 +429,7 @@ else
 fi 
 
 create_settled_icsp
+check_olm_capability
```
Comment (jianlinliu) [inline ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh:432]: @jianzhangbjz to review here.

Comment (jianzhangbjz) [reply ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh:432]: LGTM, thanks!

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-usgov-ipi-disconnected-private-f14

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-disconnected-sts-ep-fips-f28-destructive

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-f28

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-f28-destructive

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh [hunk 7]
```diff
@@ -141,23 +80,20 @@ function check_mirror_registry () {
...
 # vmc.mirror-registry.qe.devcluster.openshift.com:5000
 MIRROR_PROXY_REGISTRY=`head -n 1 "${SHARED_DIR}/mirror_registry_url"`
 echo "MIRROR_PROXY_REGISTRY: ${MIRROR_PROXY_REGISTRY}"
 set_CA_for_nodes
-# config_mirror_auth
-# When mirror images using `oc image mirror` command, need unset proxies
-unset_proxy
-mirror_tag_images
-set_proxy
 if [[ $ret -eq 0 ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:97]: I guess we do not need the if condition here.

**Comment by wewang58** [comment]
@jianlinliu  there are 4 jobs log:

**Comment by jianlinliu** [comment]
cc @mhanss @jhou1 @jianzhangbjz @LiZhang19817 @dongboyan77 to review and approve.

**Comment by wewang58** [comment]
@jhou1  seems the pr needs you to approve, if you think it's ok, can you approve it, thanks

**Comment by jianlinliu** [comment]
@anpingli @jianzhangbjz can you help approve the PR get it merged.

**Pull request merged**